rfcs/detecting_sensitive_textual_content/20230308-implementation_plan_filtering_and_designating_results_with_sensitive_textual_content.md

```diff
@@ -0,0 +1,581 @@
+                                    "fields": ["tags.name", "title", "description"],
+                                }
+                            }
+                            for term in SENSITIVE_TERMS
+                        ]
+                    }
+                },
+            },
+            "dest": {"index": destination_index},
+        },
+        slices="auto",
+        wait_for_completion=True,
+    )
+
+    self.refresh(index_name=destination_index, change_settings=True)
+
+    self.ping_callback()
+```
+
+Please note the following important details:
+
+1. `wait_for_completion` defaults to `True`, but it's worth explicitly including
+   as we rely on the fact that the request will block the process until the
+   reindexing is finished so that we know when we can promote the index.
+1. `slices` is set to `"auto"` so that Elasticsearch is free to decide the
+   optimal number of slices to use to parallelise the reindex.
+1. The source index name is not passed explicitly to the API: instead the method
+   must build it. This means we match the API of other calls (making updates to
+   the data refresh DAG much simpler) and can avoid needing to parse the model
+   name from the source index name for the call to `index_settings`.
+1. We explicitly refresh the index after it is finished reindexing to prime it
+   to be searched. ES only automatically refreshes indexes that are being
+   actively used, and the index should be refreshed before it starts being used.
+
+The creation of the new index follows the same pattern of creating the regular
+index: create the index, copy the data into the index, then issue the command to
+point the alias to the new index. This process is encapsulated in two actions:
+the `create_and_populate_filtered_index` action described above, and the
+existing `point_alias` action.
+
+```{note}
+While both processes have a "copy" step, they are notably different. The
+origin index receives its documents from Postgres via ingestion workers, which
+push the documents into Elasticsearch. The filtered index receives its
+documents via the reindex API and does not require the ingestion workers.
+```
+
+### Airflow data refresh DAG factory (overview step 2)
+
+To enable running filtered index creation independently of the data refresh
+DAGs, we will create a separate DAG that will be triggered by the data refresh
+DAGs using the
+[`TriggeredDagRunOperator`](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/trigger_dagrun/index.html).
+The new DAG will be named `<media_type>_filtered_index_creation`[^passive-name].
+It will not run on a schedule (only triggered), it will have a max concurrency
+of one, and it will accept the following configuration variables:
+
+[^passive-name]:
+    The passive voice name of this DAG is not ideal, but
+    `create_filtered_<model_name>_index_dag` reads as a DAG factory function
+    rather than a dag named "create filtered index".
```

Comment:
We don't need `_dag` in the suffix, do you think `create_filtered_<model_name>_index` works? That sounds fine IMO

Reply:
Great! I will remove this then :sweat_smile: